shell/common/shell_unittests.cc

```diff
@@ -952,5 +954,68 @@ TEST_F(ShellTest, IsolateCanAccessPersistentIsolateData) {
   DestroyShell(std::move(shell), std::move(task_runners));
 }
 
+TEST_F(ShellTest, Screenshot) {
+#if !OS_LINUX
```

Comment:
Please don't add platform specific checks for tests that check platform agnostic code. There is no reason this test should not work fine on Mac as well.

Reply:
Ah I see later you needed to do this because of goldens, you wouldn't have this problem with fixtures.

Reply:
So I did consider using fixtures, but the problem is they don't currently work on Fuchsia because the way it's set up is that there's an assumption that the build machine and the test host are the same machine (the fixtures buildsystem churns out a header file which defines the fixtures path as an absolute path on the build machine).

The goldens mechanism doesn't have this issue as we pass the path in at runtime.

I'm guessing the best way forward here is to use the fixtures path and fix it up to get rid of that assumption but I'm a little wary of adding more blockers in the way of getting tests running on Fuchsia as we have a lot already.

Reply:
You will have to account for fixtures on Fuchsia anyway because the shelL_unittests target is not going to work there without fixtures. The fixtures snapshot the Dart stuff too for example. So using goldens here doesn't mean the you can avoid having to work with fixtures to have this test target functional.

> Fuchsia because the way it's set up is that there's an assumption that the build machine and the test host are the same machine

That is only how the host targets are set up. It is not a built-in assumption. When you port fixtures to a new platform, one of the efforts is to tell test harness how to find fixtures on the platform.

> The goldens mechanism doesn't have this issue as we pass the path in at runtime.

But the assets will still have to be available at runtime on the Fuchsia device right?

Reply:
> You will have to account for fixtures on Fuchsia anyway because the shelL_unittests target is not going to work there without fixtures. The fixtures snapshot the Dart stuff too for example. So using goldens here doesn't mean the you can avoid having to work with fixtures to have this test target functional.

That's a fair point.

> > Fuchsia because the way it's set up is that there's an assumption that the build machine and the test host are the same machine
> 
> That is only how the host targets are set up. It is not a built-in assumption. When you port fixtures to a new platform, one of the efforts is to tell test harness how to find fixtures on the platform.

The fixtures gn rules seem to be platform agnostic and dumps the build directory: https://github.com/flutter/engine/blob/master/testing/testing.gni#L26

Is the plan here to make this method platform-specific so that we can specify the fixtures paths differently on each platform? I'm ok with that, just want to be clear what path forward we want to take :)

> > The goldens mechanism doesn't have this issue as we pass the path in at runtime.
> 
> But the assets will still have to be available at runtime on the Fuchsia device right?

Yep, I figured we'd either package them in the .far file or push to /tmp in run_tests.py and specify the path accordingly.